src/IECoreScene/MeshPrimitiveEvaluator.cpp

```diff
@@ -55,6 +55,73 @@ using namespace IECore;
 using namespace IECoreScene;
 using namespace Imath;
 
+namespace
+{
+
+bool triangleContainsPointV2f( const Imath::V2d& v0, const Imath::V2d& v1, const Imath::V2d& v2, const Imath::V2d& pt, Imath::V3f& bary )
+{
+	double const eps = std::numeric_limits< float >::epsilon() * 2.0;
+
+	double at = ( v0 - v1 ) % ( v2 - v0 );
+	double a1 = ( v2 - v0 ) % ( pt - v2 );
+	double a2 = ( v0 - v1 ) % ( pt - v0 );		
+
+	if( at == 0.0 )
+	{
+		return false;
+	}
+
+	if( at < 0.0 )
+	{
+		at = -at;
+		a1 = -a1;
+		a2 = -a2;
+	}
+
+	if( a2 < 0.0 )
+	{
+		if( a2 < -eps )
+		{
+			return false;
+		}
+
+		a2 = 0.0;
+	}
+
+	if( a1 < 0.0 )
+	{
+		if( a1 < -eps )
+		{
+			return false;
+		}
+
+		a1 = 0.0;
+	}
+
+	if( a2 > at )
+	{
+		if( ( a2 - at ) > ( eps ) )
+		{
+			return false;
+		}
+
+		a2 = at;
+	}
+
+	if( ( ( a1 + a2 ) - at ) > ( eps ) )
```

Comment:
I'm trying to understand the need for the lines which clamp the intermediate values of a1 and a2, on lines 88, 98, and 108. These lines don't affect the barycentric coordinate returned, since the barycentric calculation has a clamp to [0,1] on lines 116 and 117 which does the same thing ... as far as I can tell, the clamping of the intermediate values has effect only on this conditional? If this is done intentionally to changed the shape of the tolerance boundary slightly, then it definitely needs a comment here, otherwise this code would get a lot simpler by removing the clamping of the intermediate values.